habitat-lab/habitat/datasets/rearrange/rearrange_dataset.py

```diff
@@ -191,6 +197,11 @@ def to_json(self) -> str:
     @staticmethod
     def __deserialize_goal(serialized_goal: Dict[str, Any]) -> ObjectGoal:
         g = ObjectGoal(**serialized_goal)
+        # if the view points are not cached separately, read from original episodes
+        for vidx, view in enumerate(g.view_points):
+            view_location = ObjectViewLocation(**view)  # type: ignore
+            view_location.agent_state = AgentState(**view_location.agent_state)  # type: ignore
+            g.view_points[vidx] = view_location
```

Comment:
This causes an error when viewpoints are cached separately. Add a condition here.